indico/htdocs/js/indico/modules/surveys.js

```diff
@@ -53,6 +53,23 @@
         $('.js-add-question-dropdown').dropdown();
     }
 
+    global.setupSubmissionButtons = function setupSubmissionButtons() {
+        $('.js-delete-submission').on('indico:confirmed', function(e) {
+            var $this = $(this);
+            $.ajax({
+                url: $this.data('href'),
+                method: $this.data('method'),
+                complete: IndicoUI.Dialogs.Util.progress(),
+                error: handleAjaxError,
+                success: function(data) {
+                    $this.closest('li.submission-row').hide(function() {
```

Comment:
Does this fade out or something (I thought it only happens if you specify a duration in the `hide` cal)? If not simply remove it immediately (or use `.fadeOut` instead of `.hide`).

But I'd go for removing it immediately - otherwise people tend to click on the delete button again while it's fading out and sending another request, possibly resulting in an error because the submission doesn't exist anymore.